rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -537,11 +534,13 @@ impl Controller {
             api_url = api_url.to_string(),
             "Calling connlib Session::connect"
         );
+        let device_id =
+            connlib_shared::device_id::get().context("Failed to read / create device ID")?;
```

Comment:
This won't actually work until I complete the process splitting, but it compiles.